pytorch_lightning/profiler/pytorch.py

```diff
@@ -263,32 +275,405 @@ def summary(self) -> str:
-        for action, stats in recorded_stats.items():
-            output_string += (f"{os.linesep}Profile stats for: {action} rank: {local_rank} {os.linesep}{stats}")
+        local_rank = 0 if self.local_rank is None else self.local_rank
+        recorded_stats = self._summary(local_rank)
 
-        return output_string
+        output = ["Profiler Report"]
+        for action, stats in recorded_stats.items():
+            output.append(f"Profile stats for: {action} rank: {local_rank}")
+            output.append(stats)
+        return os.linesep.join(output)
 
-    def describe(self):
+    def describe(self) -> None:
         """Logs a profile report after the conclusion of the training run."""
         super().describe()
         if self.output_file:
             self.output_file.flush()
 
-    def __del__(self):
+    def __del__(self) -> None:
         """Close profiler's stream."""
         if self.output_file:
             self.output_file.close()
+
+
+class PyTorchProfiler(LegacyPyTorchProfiler):
+    """
+    This profiler uses PyTorch's Autograd Profiler and lets you inspect the cost of
+    different operators inside your model - both on the CPU and GPU.
+    Available with PyTorch>=1.8, the profiler relies on the
+    `PyTorch Kineto project <https://github.com/pytorch/kineto>`__.
+
+    The PyTorch profiler API changed in 1.8, therefore this documentation will display both versions.
+
+    Args:
+        output_filename: optionally save profile results to file instead of printing
+            to std out when training is finished. When using ``ddp``,
+            each rank will stream the profiled operation to their own file
+            with the extension ``_{rank}.txt``
+
+        enabled: Setting this to False makes this context manager a no-op.
+
+        use_cpu: Enables timing of CPU events.
+
+        use_cuda: Enables timing of CUDA events as well using the cudaEvent API.
+            Adds approximately 4us of overhead to each tensor operation.
+
+        record_shapes: If shapes recording is set, information about input dimensions will be collected.
+
+        profile_memory: Whether to report memory usage (introduced in PyTorch 1.6)
+
+        group_by_input_shapes: Include operator input shapes and group calls by shape.
+
+        with_stack: record source information (file and line number) for the ops (introduced in PyTorch 1.7)
+
+        row_limit: Limit the number of rows in a table, ``0`` is a special value that
+            removes the limit completely.
+
+        export_to_chrome: Whether to export the sequence of profiled operators for Chrome.
+            It will generate a ``.json`` file which can be read by Chrome.
+
+        path_to_export_trace: Directory path to export ``.json`` traces when using ``export_to_chrome=True``.
+            Before PyTorch 1.8, it will be save where the file being is being run.
+            After PyTorch 1.8, it will save in the ``lightning_logs/version_{}`` folder.
```

Comment:
Is this comment correct? I don't see anywhere in the code where we are adding 'lightning_logs/version_{}' to the dir path to save the trace.

Reply:
I guess it's a reference to this:

https://github.com/PyTorchLightning/pytorch-lightning/blob/5dfe46442d489d55968df7394c446bfd216215bd/pytorch_lightning/profiler/pytorch.py#L180-L183

But then it's the same before/after 1.8 and requires that a trainer.logger is present